docs/sources/experiments/architecture.md

```diff
@@ -1,3 +1,388 @@
+
+In addition, the `MatrixStore` provides a variety of methods to retrieve data from either the base matrix itself or its metadata. For instance (this is not meant to be a complete list):
+
+- `matrix` - the raw matrix
+- `metadata` - the raw metadata dictionary
+- `exists` - whether or not it exists in storage
+- `columns` - the column list
+- `labels` - the label column
+- `uuid` - the matrix's UUID
+- `as_of_dates` - the matrix's list of as-of-dates
+
+One `MatrixStorageEngine` exists at the Experiment level, and roughly corresponds with a directory wherever matrices are stored. Its only interface is to provide a `MatrixStore` object given a matrix UUID.
+
+### Model Storage
+
+Model storage is handled similarly to matrix storage, although the interactions with it are far simpler so there is no single-model class akin to the `MatrixStore`. One `ModelStorageEngine` exists at the Experiment level, configured with the Experiment's storage medium, and through it trained models can be saved or loaded. The `ModelStorageEngine` uses joblib to save and load compressed pickles of the model.
+
+### Miscellaneous Project Storage
+
+Both the `ModelStorageEngine` and `MatrixStorageEngine` are based on a more general storage abstraction that is suitable for any other auxiliary objects (e.g. graph images) that need to be stored. That is the `ProjectStorage` object, which roughly corresponds to a directory on some storage medium where we store everything. One of these exists as an Experiment attribute, and its interface `.get_store` can be used to persist or load whatever is needed.
+
+
+## Parallelization/Subclassing Details
+
+In the [Class Design](#class-design) section above, we introduced *tasks* for parallelization and subclassing for execution changes. In this section, we expand on these to help provide a new guide to working with these.
+
+Currently there are three methods that must be implemented by subclasses of Experiment in order to be fully functional.
+
+### Abstract Methods
+
+- `process_query_tasks` - Run feature generation queries. Receives a list of tasks. each `task` actually represents a table and is split into three lists of queries to enable the implementation to avoid deadlocks: `prepare` (table creation), `inserts` (a collection of INSERT INTO SELECT queries), and `finalize` (indexing). `prepare` needs to be run before the inserts and `finalize` is best run after the inserts, so it is advised that only the inserts are parallelized. The subclass should run each individual batch of queries by calling `self.feature_generator.run_commands([list of queries])`, which will run all of the queries serially, so the implementation can send a batch of queries to each worker instead of having each individual query be on a new worker. 
```

Comment:
caps, minor wording, just in the first couple lines – (each → Each, to avoid → that avoids) –

```suggestion
- `process_query_tasks` - Run feature generation queries. Receives a list of tasks. Each `task` actually represents a table and is split into three lists of queries to enable an implementation that avoids deadlocks: `prepare` (table creation), `inserts` (a collection of INSERT INTO SELECT queries), and `finalize` (indexing). `prepare` needs to be run before the inserts and `finalize` is best run after the inserts, so it is advised that only the inserts are parallelized. The subclass should run each individual batch of queries by calling `self.feature_generator.run_commands([list of queries])`, which will run all of the queries serially, so the implementation can send a batch of queries to each worker instead of having each individual query be on a new worker. 
```